lib/Parser/parsing.cpp

```diff
@@ -94,13 +95,15 @@ void Parsing::DumpCookedChars(std::ostream &out) const {
   }
 }
 
-void Parsing::DumpProvenance(std::ostream &out) const { cooked_.Dump(out); }
+void Parsing::DumpProvenance(llvm::raw_ostream &out) const {
+  cooked_.Dump(out);
+}
 
-void Parsing::DumpParsingLog(std::ostream &out) const {
+void Parsing::DumpParsingLog(llvm::raw_ostream &out) const {
   log_.Dump(out, cooked_);
 }
 
-void Parsing::Parse(std::ostream *out) {
+void Parsing::Parse(llvm::raw_ostream *out) {
```

Comment:
Is it possible to remove the pointer here also?

Reply:
The problem with this function is its initialization in parsing.h
void Parsing::Parse(llvm::raw_ostream *out = nulltptr)
And it is initialized like that because in lib/Parser/parsing.cpp:130 it calls Parse without any parameter:

Reply:
Llvm has a raw_null_ostream that you can pass instead, and you can then remove the null pointer checking code and just always print to out as it will get discarded when printing go raw_null_ostream.

Reply:
What I'm suggesting is that you could default the parameter to llvm::raw_null_ostream instead of having it as a pointer and defaulting it to null. The current solution with the pointer is probably fine though.